src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,145 @@
+            return;
+
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+        var fieldOrPropertySymbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+        if (fieldOrPropertySymbol is IPropertySymbol propertySymbol)
+        {
+            var setMethod = propertySymbol.SetMethod;
+
+            // Property must have a `set` or `init` accessor in order to be able to be required
+            if (setMethod is null)
+                return;
+
+            var containingTypeVisibility = propertySymbol.ContainingType.GetResultantVisibility();
+            var minimalAccessibility = (Accessibility)Math.Min((int)propertySymbol.DeclaredAccessibility, (int)setMethod.DeclaredAccessibility);
+
+            if (!CanBeAccessed(containingTypeVisibility, minimalAccessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_property_required, nameof(CSharpCodeFixesResources.Make_property_required));
+        }
+        else if (fieldOrPropertySymbol is IFieldSymbol fieldSymbol)
+        {
+            var containingTypeVisibility = fieldSymbol.ContainingType.GetResultantVisibility();
+            var accessibility = fieldSymbol.DeclaredAccessibility;
+
+            if (!CanBeAccessed(containingTypeVisibility, accessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_field_required, nameof(CSharpCodeFixesResources.Make_field_required));
+        }
+
+        // The `required` modifier cannot be used if a member is not accessible from outside of type.
+        // For instance, having private required property in a public class leads to compiler error.
+        // This function checks whether the member can be accessed by checking containing type visibility (which is computed already taking into account whether the type is nested and what is its visibility based on that fact)
+        // against accessibility of member we are trying to make required
+        static bool CanBeAccessed(SymbolVisibility containingTypeVisibility, Accessibility accessibility) => containingTypeVisibility switch
+        {
+            // Public is the highest accessibility. So in order to be accessible outside, member accessibility must be only public
+            SymbolVisibility.Public => accessibility is Accessibility.Public,
+            // In order to be accessible from an internal type member must have internal accessibility or higher
+            SymbolVisibility.Internal => accessibility is >= Accessibility.Internal,
+            // Private containing type visibility means it is nested in some other type.
+            // In such case member must be accessible to the outer type of containing one.
+            // This is possible with internal accessibility or higher
+            SymbolVisibility.Private => accessibility is >= Accessibility.Internal,
```

Comment:
don't understand any of this.  but as long as you've tested it well, i'm ok with it :D

Reply:
This is describing the folowwing case:
```cs
class Outer
{
    void M()
    {
        var inner = new Innner
        {
            MyProperty = ""
        }
    }

    private class Inner
    {
        public string MyProperty { get; set; }
    }
}
```
Here `SymbolVisibility` of type `Inner` is `Private` since it is a private member of an `Outer` class. However, we must be able to access the property from `Outer`, so if we declare `MyProperty` as `private`, it would cause a compiler error. That's why minimal level is internal

Reply:
why would you care about accessibility though in the requiredness checks.  It doesn't seem like it changes anything.

Reply:
Compiler is smart enough and bans the case where the accessibility or the property is lower than `internal`: https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCecAmIA1AD4ACATAAQDyArghgLABQA3q1V1QA6bIRGVSlQCSMGMxbcqHaTO58oAoelQBHOlDX5hARgAMVALI4ACugD2PDAhyyqAc1QIA3FQDOL9wF9O3PxYfIA